wdk-ddi-src/content/fltkernel/ns-fltkernel-_flt_parameters.md

```diff
@@ -105,6 +105,11 @@ The following I/O operations do not have parameters, and therefore do not have a
 - IRP_MJ_SHUTDOWN
 - IRP_MJ_VOLUME_DISMOUNT
 
+For the operations that come in Acquire/Release pairs (ModWrite, CcFlush, Section), there is no guarantee that the release call will happen even after a sucessfull call to the post acquire callback.  An example would be a if an instance detach occurs.
+
+Counterinutitively, you should therefore avoid acquring resources in the Acquire call and releasing them in the release call.  If you have to you should either stall the instance detach (*before* Instance TearDownStart callback) or arrange for the resources to be released during instance tear down.

```

Comment:
I would strike this paragraph since this is true for any callback. It is a general rule that a filter should not acquire resources and then leave them held after exiting a callback. Doing what is described here is even worse than acquiring a resource in a pre-operation callback and releasing it in a post-operation callback. This describes getting the pre- and post-operation callbacks for a particular operation (e.g. `IRP_MJ_ACQUIRE_FOR_SECTION_SYNCHRONIZATION`) and then _leaving them held_ until some unknown future time.